### PR TITLE
[vesper] Add Vesper strategy

### PR DESCRIPTION
**Pull request by marcelomorgado**
This strategy calls the `VesperVotingPower` in order to get accounts' voting power.

### CODE REVIEW

File: src/strategies/index.ts [hunk 1]
```diff
@@ -348,12 +349,14 @@ const strategies = {
   'mutant-cats-stakers-and-holders': mutantCatsStakersAndHolders,
   'svs-staking': svsStaking,
   'mcb-balance-from-graph': mcbBalanceFromGraph,
+  'colony-reputation': colonyReputation,
```
Comment (ChaituVR) [inline src/strategies/index.ts:352]:
```suggestion
  'mcb-balance-from-graph': mcbBalanceFromGraph,
```

File: src/strategies/vesper/index.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import { formatUnits } from '@ethersproject/units';
+import { getBlockNumber } from '../../utils';
+import { Multicaller } from '../../utils';
+
+export const author = 'Vesper Finance';
```
Comment (ChaituVR) [inline src/strategies/vesper/index.ts:5]:
```suggestion
export const author = 'marcelomorgado';
```

File: src/strategies/vesper/index.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      break;
+    }
+    snapshotBlocks.push(blockTag);
+  }
+
+  const scores = await Promise.all([
+    ...snapshotBlocks.map((blockTag) =>
+      getScores(provider, addresses, options, blockTag)
+    )
+  ]);
```
Comment (ChaituVR) [inline src/strategies/vesper/index.ts:34]: @marcelomorgado I think we should have a limit to `periods` param else if someone passes thousands of previous blocks, strategy won't scale 
ideally, it should be like `5` maximum blocks maybe?


The limit of the number of requests/multicall is `5` 
https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/vesper/index.ts (hunk 3 above)
Comment (marcelomorgado) [reply src/strategies/vesper/index.ts:36]: Done. I've hardcoded `period = 4`

File: src/strategies/vesper/index.ts (hunk 2 above)
Comment (marcelomorgado) [reply src/strategies/vesper/index.ts]: Done.

File: src/strategies/index.ts (hunk 1 above)
Comment (marcelomorgado) [reply src/strategies/index.ts]: Done.

**Review by bonustrack — APPROVED** [review]

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**